automerge-backend/src/object_store.rs

```diff
@@ -188,6 +229,28 @@ impl MapState {
         }
     }
 
+    fn generate_diffs(&self) -> Vec<Diff> {
+        let mut diffs = vec![];
+        if self.object_id != ObjectID::Root {
+            diffs.push(
+                Diff {
+                    action: DiffAction::CreateMap(self.object_id.clone(), self.map_type.clone()),
+                    conflicts: vec![],
```

Comment:
I've been tending to use `Vec::new()` rather than `vec![]` for creating empty arrays, no idea which is better but we should probably choose one for consistency.